src/DynamoCore/Models/NodeModel.cs

```diff
@@ -1669,8 +1668,32 @@ private void OnRenderPackageUpdateCompleted(AsyncTask asyncTask)
             var task = asyncTask as UpdateRenderPackageAsyncTask;
             if (task.RenderPackages.Any())
             {
-                OnRenderPackagesUpdated(task.RenderPackages);
+                var packages = new List<IRenderPackage>();
+                
+                packages.AddRange(task.RenderPackages);
+                packages.AddRange(OnRequestRenderPackages());
+
+                OnRenderPackagesUpdated(packages);
+            }
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public event Func<IEnumerable<IRenderPackage>> RequestRenderPackages;
+
+        /// <summary>
+        /// This event handler is invoked when the render packages (specific to this node)  
+        /// become available and in addition the node requests for associated render packages 
+        /// if any for example, packages used for associated node manipulators
+        /// </summary>
+        private IEnumerable<IRenderPackage> OnRequestRenderPackages()
+        {
+            if (RequestRenderPackages != null)
+            {
+                return RequestRenderPackages();
             }
+            return new List<IRenderPackage>();
```

Comment:
Raises the event that requests render packages for any manipulators associated with the node. This event is registered with the `DrawManipulator` method in `Manipulator` base class in DM project. See PR: https://git.autodesk.com/Dynamo/DynamoPro/pull/178